chain-test/src/test/scala/org/bitcoins/chain/util/ChainUnitTest.scala

```diff
@@ -59,6 +67,65 @@ trait ChainUnitTest
     dropHeaderTable(testExecutionF)
   }
 
+  def withPopulatedBlockHeaderDAO(
+      test: BlockHeaderDAO => Future[Assertion]): Future[Assertion] = {
+    val tableSetupF = setupHeaderTable()
+
+    val source =
+      scala.io.Source.fromURL(getClass.getResource("/block_headers.json"))
+    val arrStr = source.getLines.next
+    source.close()
+
+    import org.bitcoins.rpc.serializers.JsonReaders.BlockHeaderReads
+    val headersResult = Json.parse(arrStr).validate[Vector[BlockHeader]]
+
+    headersResult match {
+      case err: JsError =>
+        logger.error(s"Failed to parse headers from block_headers.json: $err")
+        Future.failed(new RuntimeException(err.toString))
+      case JsSuccess(headers, _) =>
+        val dbHeaders = headers.zipWithIndex.map {
+          case (header, height) =>
+            BlockHeaderDbHelper.fromBlockHeader(height, header)
+        }
+
+        @tailrec
+        def splitIntoBatches(
```

Comment:
so just an FYI, when we are doing initial block download (IBD) we do get headers in batches of 2,000 headers. We should probably make sure that it is isn't terrible for the database to handle 2,000 header batches, and if it is, we might want to consider moving this into the `BlockHeaderDAO` createAll method to batch things

https://bitcoin.org/en/developer-reference#getheaders